File: Java/LongestSubstringWithoutRepeatingCharacters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int longest = 0;
        String current = "";
        
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            int index = current.indexOf(c);
            if (index != -1) {
                current = current.substring(index + 1);
            }
            current += c;
            if (current.length() > longest) {
                longest = current.length();
            }
        }
        return longest;
    }
}

File: Python/LongestSubstringWithoutRepeatingCharacters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """

        current = ""
        length = len(s)
        longest = 0

        for i in range(length):
            c = s[i]
            index = 0
            if c in current:
                index = current.index(c)
            else:
                index = -1
            if index != -1:
                current = current[index + 1:]
            current += c
            if len(current) > longest:
                longest = len(current)
        return longest